llvm/include/llvm/Support/GenericLoopInfoImpl.h

```diff
@@ -208,7 +208,10 @@ BlockT *LoopBase<BlockT, LoopT>::getLoopPreheader() const {
     return nullptr;
 
   // Make sure there is only one exit out of the preheader.
-  if (llvm::size(llvm::children<BlockT *>(Out)) != 1)
+  typedef GraphTraits<BlockT *> BlockTraits;
+  typename BlockTraits::ChildIteratorType SI = BlockTraits::child_begin(Out);
+  ++SI;
+  if (SI != BlockTraits::child_end(Out))
```

Comment:
Could we do something like this?

```suggestion
  if (!llvm::hasSingleElement(GraphTraits<BlockT *>::children(Out)))
```

Reply:
Ah nice, much better. Switched to `llvm::hasSingleElement`, thanks!